client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -22,11 +37,333 @@ varying mediump vec2 varTexCoord;
 centroid varying vec2 varTexCoord;
 #endif
 varying vec3 eyeVec;
+varying float nightRatio;
 
 const float fogStart = FOG_START;
 const float fogShadingParameter = 1.0 / ( 1.0 - fogStart);
 
-#if ENABLE_TONE_MAPPING
+
+#ifdef ENABLE_DYNAMIC_SHADOWS
+	
+	#ifdef SHADOWS_PSM
+    const float bias0 = 0.94;
+    
+    const float zPersFactor = 0.25;
+
+    vec4 getPerspectiveFactor(in vec4 shadowPosition) {
+      float bias1 = 1.0 - bias0;
+      float pDistance =  sqrt(shadowPosition.x * shadowPosition.x +
+          shadowPosition.y * shadowPosition.y );
+      float pFactor = pDistance * bias0 + bias1;
+      shadowPosition.xyz *= vec3(vec2(1.0 / pFactor), zPersFactor);
+
+      return shadowPosition;
+    }
+	#endif
```

Comment:
You should also aim for a consistent code style for GLSL files, currently the spacing and indentation here is pretty random.
Also if the commented out code (various places) is not needed anymore I suggest removing it.

Reply:
done.